src/robot/libraries/BuiltIn.py

```diff
@@ -2232,7 +2244,16 @@ def wait_until_keyword_succeeds(self, retry, retry_interval, name, *args):
                     raise AssertionError("Keyword '%s' failed after retrying "
                                          "%s. The last error was: %s"
                                          % (name, message, err))
-                self._sleep_in_parts(retry_interval)
+                keyword_runtime = time.time() - start_time
+                if strict_interval:
+                    sleep_time = retry_interval - keyword_runtime
+                if sleep_time < 0:
+                    logger.warn("Interval violation: retry_interval is {}"
+                                ", but keyword runtime is {}."
+                                .format(secs_to_timestr(retry_interval),
+                                        secs_to_timestr(keyword_runtime)))
```

Comment:
This warning still isn't tested.

We also have used the old style `%` formatting elsewhere and for consistency it would be better here as well. Looking forward to the (near) future when we can just use f-strings!